drivers/tee/tee_core.c

```diff
@@ -454,25 +582,53 @@ static int tee_ioctl_invoke(struct tee_context *ctx,
 		rc = params_from_user(ctx, params, arg.num_params, uparams);
 		if (rc)
 			goto out;
+
+		/*
+		 * The OCALL parameter must be first, or not be present at all.
+		 * This function returns an error if the OCALL parameter is
+		 * found in the wrong place, a pointer to the parameter if it is
+		 * present, and NULL otherwise. Hence, 'ocall' can still be NULL
+		 * after this statement.
+		 */
+		ocall = tee_param_find_ocall(params, arg.num_params);
```

Comment:
We're looping through all params to check that stuff is in order. How about also returning: normal_params, num_normal_params, ocall_params, num_ocall_params and then pass that directly to `ctx->teedev->desc->ops->invoke_func()` below?